### DISCUSSION TITLE
Form object validation

### DISCUSSION THREAD

**Discussion by azamspeare** [livewire/livewire#6743] [1. Help]
I am using Livewire3 for model crud. And I have used Form object class to separate properties.

```
class LanguageForm extends Form
{
    public Language $language;

    #[Rule('required|string|min:2|unique:languages,slug')]
    public ?string $slug = null;

    #[Rule('required|string|min:2')]
    public ?string $title = null;
```

I am check is slug unique or not. In creating I have to check but, in updating existing language I don't need to check is unique. I have also disabled this field, but anyway when try to update it checks is unique or not. How can I make flexible rule for both updating and creating?

Create component:
```
class Create extends Component
{
    public LanguageForm $form;

    public function create(): void
    {
            $this->validate();
    }
```

Edit component:
```
class Edit extends Component
{
    public LanguageForm $form;

    public function mount(Language $item): void
    {
        $this->form->setValues($item);
    }

    public function edit(): void
    {
        try {
            $this->validate();
        }
```

**Comment by muhdazrain** [comment]
Hi. I also encountered similar to your problem when to validate unique. As far as I know, you need to use Rule object in the validation. However, rule attribute does not support for this and you need to define rules() in the component.

If you think your form just need simple validation, then you can use new Form object feature. But if you think need some advanced validation, I recommend no need to use that Form object and just use like in V2.

Thanks.

**Comment by krekas** [comment]
You cannot use attributes for this rule. In you component you must have a model property and can access it in the form object using `$this->component`. Example rule:

```php
public function rules(): array
{
    return [
        'name'        => [
            'required',
            Rule::unique('products', 'name')->ignore($this->component->product), 
        ],
    ];
}
```
